### PR TITLE
Webpack build process

### PR DESCRIPTION
**Pull request by jmweir**
Supports #32.

Webpack bundles a set of source files that may employ Node-based libraries into a set of distribution files ready to be served on the web. Because it transforms multiple interdependent sources, it's easier to separate various concerns. This is the new source tree:
  - bookmarklet.html : a sample bookmarklet link suitable for debug purposes
  - index.js : entry point that ties everything together
  - summary-style.css : summary container stylesheet
  - summary-template.tpl : summary container template
  - template.js : a simple templating library
  - unanet-summarizer.js : a function for generating a summary of a Unanet timesheet

### CODE REVIEW

**Comment by SeanKilleen** [comment]
@jmweir I do like this direction. I think before we could accept a PR that introduces something that needs a build/publish process, I'd need to know a transition path for current users.

* Users currently use a bookmarklet that calls to a specific URL
* we'll likely need to publish elsewhere
* I'd like to avoid having to commit the output of a build process anywhere.
* It's still important to me that we keep the build output super small. I'd like to avoid giant dependencies and keep it simple.

One thing we could do to remain backwards compatible, I was thinking: 

* the bookmarklet could load our current script
* the contents of our current script could be changed to load the real script
* we can then publish the script to a blob or bucket

What do you think?

Also -- If we do that, we may want to look into the ramifications of appearing to load a strange script vs one from excellalabs.github.io (security scans on client sites, etc.)

**Comment by jmweir** [comment]
Good points! Quick thoughts:
* Built JS file is only 5k
* No dependencies have been added here for production requirement
* Agreed on thinking more through the build/publish process. Committing the build output is a temporary measure to achieve backward compatibility.

**Comment by SeanKilleen** [comment]
@jmweir do you think that with my suggestion above, we could achieve this in small steps?

PR 1: move files and make our script load another script -- so we have the redirection capability in place without disrupting bookmarklet users (we can check it locally that way too)

PR 2: create/add files for build process

PR 3: set up pipeline and output to somewhere

PR 4: change the redirect link to point to the new build output and delete the file that's in the repo

**Comment by SeanKilleen** [comment]
@jmweir webpack looks great so far!

Next steps on this PR as I see it:

* output it to a `dist` folder` so we can set up a build / release pipeline later
* revert the changes to `unanet-summarizer.js` (you'll need to merge in master and let its changes overrules these ones).
* lets add a chunk of text to the README as part of the PR that explains the components as you describe them, and the process for building.

File: src/bookmarklet.html [hunk 1]
```diff
@@ -0,0 +1 @@
+<a href="javascript:(function(){ var onload=function(){ window.summarizeUnanetTimeForReal(); }; if(!window.summarizeUnanetTimeForReal){var s=document.createElement('script');s.src='http://localhost:8080/unanet-summarizer-release.js';s.onload=onload;document.body.appendChild(s)} else { onload(); } })();">Summarize Unanet (Debug)</a>
```
Comment (SeanKilleen) [inline src/bookmarklet.html]: Nice touch. 👍

**Comment by SeanKilleen** [comment]
@jmweir FYI I took a sec and added `README` draft contents in your PR to move it forward there. Feel free to change or accept as-is.

**Comment by jmweir** [comment]
@SeanKilleen I think this is good to go now. Leaving for a final check by you. Feel free to merge if you're good with it.

File: README.md [hunk 2]
```diff
@@ -43,7 +43,7 @@ This bookmarklet adds injects the JavaScript file if it doesn't exist, and then
 
 ### How to test locally
 
-* `npm run serve` allows you to view a bookmarklet for debug that you drag to your bookmarks bar
+* `npm start` allows you to view a bookmarklet for debug that you drag to your bookmarks bar
```
Comment (SeanKilleen) [inline README.md]: should this be `npm run start`?

**Comment by jmweir** [comment]
It is "npm start" in this case.

File: README.md (hunk 2 above)
Comment (SeanKilleen) [reply README.md]: Nevermind this totally works 👍

**Comment by SeanKilleen** [comment]
@jmweir really awesome work on this. Very excited to have this here and looking forward to hooking it up to a build process ASAP.

**Pull request merged**